src/compiler/utilities.ts

```diff
@@ -3298,6 +3297,10 @@ namespace ts {
         return false;
     }
 
+    export function isJSDocTypeReference(node: TypeReferenceType): node is TypeReferenceNode {

```

Comment:
Yep, I realised that last night. In fact, `finishNode` in the parser persists the context flag, so it was there all along.